packages/grant-explorer/src/features/discovery/RoundDaysDetails.tsx

```diff
@@ -0,0 +1,76 @@
+import tw from "tailwind-styled-components";
+
+export function getRoundDaysText({
+  roundStartsIn,
+  roundEndsIn,
+}: {
+  roundStartsIn?: number;
+  roundEndsIn?: number;
+}) {
+  const days = pluralize(["day", "days"]);
+
+  if (roundStartsIn && roundStartsIn > 0) {
+    return `Starts in ${roundStartsIn} ${days(roundStartsIn)}`;
+  }
+  if (roundEndsIn === 0) return "Ends today";
+  if (!roundEndsIn) return "No round end date";
+
+  return roundEndsIn > 0
+    ? `${roundEndsIn} ${days(roundEndsIn)} left in round`
+    : `Ended ${-roundEndsIn} ${days(-roundEndsIn)} ago`;
+}
+
+export function getRoundApplicationDaysText({
+  applicationsStartsIn,
+  applicationsEndsIn,
+}: {
+  applicationsStartsIn?: number;
+  applicationsEndsIn?: number;
+}) {
+  const days = pluralize(["day", "days"]);
+
+  // Hide if application date has passed
+  if (applicationsEndsIn === undefined || applicationsEndsIn < 0) return "";
+
+  if (applicationsEndsIn === 0) return "Last day to apply";
+  if (
+    applicationsStartsIn &&
+    applicationsStartsIn > 0 &&
+    applicationsEndsIn > 0
+  ) {
+    return `Apply in ${applicationsStartsIn} ${days(applicationsStartsIn)}`;
+  }
+  return `${applicationsEndsIn} ${days(applicationsEndsIn)} left to apply`;
```

Comment:
Only mentioning this kicks in case you like pattern matching: in TypeScript 5.3 it will be finally possible to have:

```ts
switch(true) {
  case applicationsEndsIn === undefined: 
    return "";
  case applicationsEndsIn < 0:
    return "";
  case applicationsEndsIn === 0: 
    return "Last day to apply";
  // ...
  default:
    return `${applicationsEndsIn} ${days(applicationsEndsIn)} left to apply`;
}
```

Reply:
Interesting! Thanks for sharing